src/Models/ProductAttribute.php

```diff
@@ -0,0 +1,172 @@
+        'date' => 'Date',
+        'boolean' => 'Yes/No',
+        'multiselect' => 'Multiple Select',
+        'select' => 'Dropdown',
+        'price' => 'Price',
+        'media_image' => 'Media Image',
+        'weee' => 'Fixed Product Tax',
+        'swatch_visual' => 'Visual Swatch',
+        'swatch_text' => 'Text Swatch',
+    ];
+
+    protected static function booting()
+    {
+        static::addGlobalScope(function (Builder $builder) {
+            $builder->where('entity_type_id', function ($query) {
+                $query->select('entity_type_id')
+                    ->from('eav_entity_type')
+                    ->where('entity_type_code', 'catalog_product');
+            });
+
+            $builder->leftJoin('eav_attribute_label', function ($join) {
+                $join->on('eav_attribute.attribute_id', '=', 'eav_attribute_label.attribute_id')
+                    ->where('eav_attribute_label.store_id', static::getCurrentStoreId());
+            });
+
+            $builder->leftJoin('eav_attribute_option', 'eav_attribute.attribute_id', '=', 'eav_attribute_option.attribute_id')
+                ->leftJoin('eav_attribute_option_value as admin_value', function ($join) {
+                    $join->on('eav_attribute_option.option_id', '=', 'admin_value.option_id')
+                        ->where('admin_value.store_id', 0);
+                })
+                ->leftJoin('eav_attribute_option_value as store_value', function ($join) {
+                    $join->on('store_value.option_id', '=', 'eav_attribute_option.option_id')
+                        ->where('store_value.store_id', static::getCurrentStoreId());
+                });
+
+            $builder->select([
+                'eav_attribute.*',
+                'eav_attribute_label.value as store_frontend_label',
+                DB::raw('GROUP_CONCAT(DISTINCT COALESCE(store_value.value, admin_value.value)) as option_values'),
+                DB::raw('GROUP_CONCAT(DISTINCT eav_attribute_option.option_id) as option_ids')
+            ])
+            ->groupBy([
+                'eav_attribute.attribute_id',
+                'eav_attribute.attribute_code',
+                'eav_attribute.frontend_input',
+                'eav_attribute.frontend_label',
+                'eav_attribute_label.value'
+            ]);
+        });
+    }
+
+   public function options(): Attribute
+   {
+       return Attribute::make(
+           get: fn () => !$this->option_ids || !$this->option_values ? [] : 
+               (count($optionIds = array_filter(explode(',', $this->option_ids))) === count($optionValues = array_filter(explode(',', $this->option_values))) ? array_combine($optionIds, $optionValues) : [])
+       );
+   }
+
+   public function formattedOptions(): Attribute
+   {
+       return Attribute::make(
+           get: fn () => empty($this->options) ? '' : 
+               collect($this->options)
+                   ->map(fn($value, $id) => "{$value} (ID: {$id})")
```

Comment:
Can't we use single quotes here?

Reply:
I'd rather keep it like this, keep it more compact